extra/CMakeLists.txt

```diff
@@ -73,7 +73,9 @@ else()
 endif()
 
 if(GUROBILIB)
-	set(GUROBI_LIB "-L${GUROBI_LIB_DIR} -lgurobi${GUROBI_VERSION} -lm -lpthread")
+	get_filename_component(LIBGRB ${GUROBILIB} NAME_WE)
```

Comment:
What do we get in `GUROBILIB` at this line? I am just curious.

Reply:
It is the full path of the Gurobi library. On my laptop is "/Library/gurobi901/mac64/lib/libgurobi90.dylib"

Reply:
OK, then it should be enough to replace lines 76-78 with
```cmake
set(GUROBI_LIB "-l${GUROBILIB} -lm -lpthread")
```
Does gurobi also need to link `m` and `pthread`?

Reply:
I am pretty sure m is needed. But pthread is sometimes needed, depending on OS / compiler.

Reply:
> OK, then it should be enough to replace lines 76-78 with
> 
> ```cmake
> set(GUROBI_LIB "-l${GUROBILIB} -lm -lpthread")
> ```
> 
> Does gurobi also need to link `m` and `pthread`?
I also tried this. I got errors with this command
```
checking if user provides library for Gurobi... yes
configure: error: Cannot find file gurobi_c.h in /Users/yingqiuzhang
configure: error: /bin/sh '/Users/yingqiuzhang/Documents/DSPtest/DSP/extra/Osi/Osi/configure' failed for Osi
make[2]: *** [extra/Osi-prefix/src/Osi-stamp/Osi-configure] Error 1
make[1]: *** [extra/CMakeFiles/Osi.dir/all] Error 2
make: *** [all] Error 2

```

Reply:
It seems related to this line.
```cmake
set(WITH_GUROBI_INCDIR "--with-gurobi-incdir=${GUROBI_INC_DIR}")
```

Reply:
> It seems related to this line.
> 
> ```cmake
> set(WITH_GUROBI_INCDIR "--with-gurobi-incdir=${GUROBI_INC_DIR}")
> ```


Sorry that I did not reproduce my error correctly, it is actually like this

```
configure: error: Cannot find symbol(s) GRBloadenv with GRB
configure: error: /bin/sh '/Users/yingqiuzhang/Documents/DSPtest/DSP/extra/Osi/Osi/configure' failed for Osi
make[2]: *** [extra/Osi-prefix/src/Osi-stamp/Osi-configure] Error 1
make[1]: *** [extra/CMakeFiles/Osi.dir/all] Error 2
make: *** [all] Error 2
```